example-widget-mui/src/DicePage/DicePage.tsx

```diff
@@ -97,9 +107,54 @@ export const DiceView = (): ReactElement => {
     setLastOwnDice(result.content.pips);
   }
 
+  async function handleThrowDiceDelayed() {
+    await widgetApi.requestCapabilities([
+      MatrixCapabilities.MSC4157SendDelayedEvent,
+      WidgetEventCapability.forRoomEvent(
+        EventDirection.Send,
+        STATE_EVENT_THROW_DICE,
+      ),
+    ]);
+
+    const pips = Math.floor(Math.random() * 6) + 1;
+    try {
+      const { delay_id } = await widgetApi.sendDelayedRoomEvent<ThrowDiceEvent>(
+        STATE_EVENT_THROW_DICE,
+        { pips },
+        10000,
+      );
+      setLastOwnDice(pips);
+      setLastDelayId(delay_id);
+      setLastDelayError(undefined);
+    } catch {
+      setLastDelayError(
+        'Could not send a delayed event. Please check if homeserver supports delayed events.',
+      );
+    }
+  }
+
+  async function handleThrowDiceDelayedUpdate(
+    action: UpdateDelayedEventAction,
+  ) {
+    if (!lastDelayId) {
+      return;
+    }
+
+    await widgetApi.requestCapabilities([
+      MatrixCapabilities.MSC4157UpdateDelayedEvent,
+    ]);
+
+    try {
+      await widgetApi.updateDelayedEvent(lastDelayId, action);
```

Comment:
This method succeeding implies success of the requested action. I think it is part of a complete example to demonstrate that this case needs to be handled. Handling needs to be differently depending on the action:
- cancel and send implies some UI cleanup (don't offer to cancel/send again)
- by resend we stay in this state.

Reply:
Okay, `Dice` example is updated.